packages/cli/src/services/public-api-key.service.ts

```diff
@@ -0,0 +1,84 @@
+import { randomBytes } from 'crypto';
+import Container, { Service } from 'typedi';
+
+import { ApiKey } from '@/databases/entities/api-key';
+import type { User } from '@/databases/entities/user';
+import { ApiKeyRepository } from '@/databases/repositories/api-key.repository';
+import { UserRepository } from '@/databases/repositories/user.repository';
+
+export const API_KEY_PREFIX = 'n8n_api_';
+
+@Service()
+export class PublicApiKeyService {
+	constructor(private readonly apiKeysRepository: ApiKeyRepository) {}
+
+	/**
+	 * Creates a new public API key for the specified user.
+	 *
+	 * @param {User} user - The user for whom the API key is being created.
+	 * @returns {Promise<ApiKey>} A promise that resolves to the newly created API key.
+	 */
+	async createPublicApiKeyForUser(user: User) {
+		const apiKey = this.createApiKeyString();
+		await this.apiKeysRepository.upsert(
+			this.apiKeysRepository.create({
+				userId: user.id,
+				apiKey,
+				label: 'My API Key',
```

Comment:
What if I create two api keys? The second one fails, right? Or is this something handled in the next PRs?

Reply:
This will be addressed when we allow in the UI to create multiples ones, and ask for the label.